src/hotspot/share/cds/lambdaFormInvokers.cpp

```diff
@@ -79,12 +79,22 @@ void LambdaFormInvokers::append(char* line) {
   _lambdaform_lines->append(line);
 }
 
-void LambdaFormInvokers::regenerate_holder_classes() {
+#define HANDLE_IF_HAS_EXCEPTION                                                                         \
+  if (HAS_PENDING_EXCEPTION) {                                                                          \
+    if (!PENDING_EXCEPTION->is_a(vmClasses::OutOfMemoryError_klass())) {                                \
```

Comment:
I would suggest adding a comment and error message to explain why we ignore the error:

```
// We may get an exception if the classlist contains an error (or an outdated entry generated by
// an older JDK).
if (DumpSharedArchive) {
  log_error(cds)("Failed to generate LambdaForm holder classes. Is your classlist out of date?");
} else {
  log_error(cds)("Failed to generate LambdaForm holder classes. Was the base archive generated with an outdated classlist?");
}
```

Reply:
Do you mean DumpSharedSpaces? So the message will be different between static (DumpSharedSpaces) and dynamic, right?

Reply:
Yes, because the errors can only be caused by bad input from the classlist, which is only used during static archive dump. The information collected during dynamic dump are correct, and cannot cause `CDS.generateLambdaFormHolderClasses()` to fail.